.github/workflows/soakTest.yml

```diff
@@ -26,124 +30,164 @@ jobs:
+      - name: Clone adapter-secrets repo
         env:
-          ADAPTER_NAMES: ${{ steps.get-pr-info.outputs.ADAPTERS_LIST }}
+          GITHUB_TOKEN: ${{ secrets.QA_GITHUB_TOKEN }}
+        run: git clone https://${GITHUB_TOKEN}@github.com/smartcontractkit/adapter-secrets.git
+
+      - name: Build the k6 payloads and images
+        id: k6-payloads
+        env:
+          PR_NUMBER: ${{ steps.findPr.outputs.pr }}
+          TEST_ADAPTERS: ${{ steps.get-pr-info.outputs.TEST_ADAPTERS }}
+          IMAGE_PREFIX: ${{ secrets.QA_AWS_ACCOUNT_NUMBER }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/k6
         run: |
-          set -x
-          ADAPTERS_LIST=
-          for adapter in ${ADAPTER_NAMES}; do
-            echo "${adapter}"
-            DONT_ADD=0
-            yarn qa:flux:configure exists ${adapter} blank || DONT_ADD=1
-            if [ $DONT_ADD = '1' ]; then
-              continue
-            fi
-            ADAPTERS_LIST="${ADAPTERS_LIST} ${adapter}"
-          done
-          ADAPTERS_LIST="$(echo ${ADAPTERS_LIST} | xargs)"
-          # export adapters from this step
-          echo "::set-output name=ADAPTERS_LIST::${ADAPTERS_LIST}"
+          if [ -n "$TEST_ADAPTERS" ]; then
+            for adapter in ${TEST_ADAPTERS}; do
+              yarn qa:flux:configure k6payload ${adapter} ${PR_NUMBER} || exit 1
+
+              cd ./packages/k6
+
+              echo "" >> ./.env
+              echo "PAYLOAD_GENERATED=true" >> ./.env
+              echo "CI_ADAPTER_NAME=${adapter}" >> ./.env
+              echo "QA_RELEASE_TAG=${PR_NUMBER}" >> ./.env
+              echo "PR_NUMBER=${PR_NUMBER}" >> ./.env
+              echo "TEST_DURATION=10m" >> ./.env
+              cat ./.env
+
+              yarn build
+              docker build -t ${IMAGE_PREFIX}:pr${PR_NUMBER}-${adapter} .
+              docker push ${IMAGE_PREFIX}:pr${PR_NUMBER}-${adapter}
+
+              cd ../..
+            done
+          fi
+
       - name: Build adapters
+        id: build-adapters
         env:
-          PR_NUMBER: ${{ steps.findPr.outputs.pr }}
-          ADAPTER_NAMES: ${{ steps.adapters-list.outputs.ADAPTERS_LIST }}
-          IMAGE_PREFIX: ${{ secrets.SDLC_ACCOUNT_ID }}.dkr.ecr.${{ secrets.AWS_REGION }}.amazonaws.com/adapters/
+          TEST_ADAPTERS: ${{ steps.get-pr-info.outputs.TEST_ADAPTERS }}
+          IMAGE_TAG: pr${{ steps.findPr.outputs.pr }}
+          IMAGE_PREFIX: ${{ secrets.QA_AWS_ACCOUNT_NUMBER }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/adapters/
         run: |
-          IMAGE_TAG=pr${PR_NUMBER}
-          IMAGE_TAG=${IMAGE_TAG} IMAGE_PREFIX=${IMAGE_PREFIX} yarn generate:docker-compose
+          GITHUB_WORKSPACE= IMAGE_TAG=${IMAGE_TAG} IMAGE_PREFIX=${IMAGE_PREFIX} yarn generate:docker-compose
+
+          BUILD_ADAPTERS=
+          if [ -n "$TEST_ADAPTERS" ]; then
+            yarn qa:dependencies $TEST_ADAPTERS
+            DEP=$(cat ./packages/scripts/src/adapter-dependencies/dependencies.txt)
```

Comment:
Maybe we add this to documentation, that dependencies need to be updated here for new/updated EAs

Reply:
The `yarn qa:dependencies` script should automatically update the dependency list (`packages/scripts/src/adapter-dependencies/lib.ts`).

I've added to the documentation ticket that we need to document things like updating the helmcharts.